src/features/rewards/mobile/selectMobile/index.tsx

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import {
+  StyledSelect
+} from './style'
+import { getLocale } from '../../../../helpers'
+
+export type Amount = {
+  value: string
+  dataValue: string
+  converted: string
+}
+
+export type Option = {
+  value: string
+  text: string
+}
+
+export interface Props {
+  value?: string
+  options?: Option[]
+  floating?: boolean
+  amountOptions?: Amount[]
+  onChange?: (value: string) => void
+}
+
+export default class SelectMobile extends React.PureComponent<Props, {}> {
+
+  onChange = (event: any) => {
+    if (this.props.onChange) {
+      this.props.onChange(event.target.value)
+    }
+  }
+
+  generateOptions = (value?: string, options?: Option[]) => {
+    if (!options) {
+      return null
+    }
+
+    return (
+      <>
+        {options.map((option: Option) => {
+          return (
+            <option
+              value={option.value}
+              key={`k-${option.value}`}
+              selected={option.value === value}
+            >
+              {option.text}
+            </option>
+          )
+        })}
+      </>
+    )
+  }
+
+  generateAmountOptions = (value?: string, amountOptions?: Amount[]) => {
+    if (!amountOptions) {
+      return null
+    }
+
+    return (
+      <>
+        {amountOptions.map((amount: Amount) => {
+          return (
+            <option
+              key={`k-${amount.value}`}
+              value={amount.dataValue}
+              selected={amount.dataValue === value}
+            >
+              {amount.value} {getLocale('bat')} ({amount.converted} USD)
+            </option>
+          )
+        })}
+      </>
+    )
+  }
+
+  render () {
+    const { options, floating, amountOptions, value } = this.props
```

Comment:
We don't totally want to omit floating, because it also determines whether we have a border around select or not.

For example, in auto-contribute settings the select boxes have a border, but in the detail view for monthly budget aligned on the right, we want that to be borderless per design, unless you are fine with the alternative

Reply:
👍